src/classes/Application/Traits/Admin/Wizard.php

```diff
@@ -610,29 +665,42 @@ public function getURL(array $params=array()) : string
     public function handle_stepUpdated(Application_Admin_Wizard_Step $updatedStep) : void
     {
         $number = $updatedStep->getNumber();
-        foreach($this->steps as $step) {
-            if($step->getNumber() > $number && $step->isMonitoring($updatedStep)) {
+        foreach ($this->steps as $step)
+        {
+            if ($step->getNumber() > $number && $step->isMonitoring($updatedStep))
+            {
                 $step->handle_stepUpdated($updatedStep);
             }
         }
+        if ($this->invalidationHandler->isInvalidated() && $number === $this->invalidationHandler->getInvalidationCallingStep())
+        {
+            $this->redirectWithErrorMessage($this->invalidationHandler->getInvalidationMessage(), $this->invalidationHandler->getInvalidationURL());
+        }
     }
-    
+
     /**
      * Handles a step being invalidated: updates all steps that are
      * dependent on this step, saves all step data, and redirects
      * to the step so the user can complete it anew.
      *
      * @param Application_Admin_Wizard_Step $step
      * @param string $reasonMessage
+     * @param int $callingStep
      */
-    public function handle_stepInvalidated(Application_Admin_Wizard_Step $step, string $reasonMessage) : void
+    public function handle_stepInvalidated(Application_Admin_Wizard_Step $step, string $reasonMessage, int $callingStep = 0) : void
     {
         $this->handle_stepUpdated($step);
         $step->setComplete(false);
         $this->saveSettings();
-        $this->redirectWithErrorMessage($reasonMessage, $step->getURL());
+        if (!$this->invalidationHandler->isInvalidated())
+        {
+            $this->invalidationHandler->setInvalidationMessage($reasonMessage);
```

Comment:
Move this to its own method, so it's easy to find.

Additionally, consider changing the invalidation handler class to support chaining the methods, so the code can be made easier to read.